### PR TITLE
ignore generated weaver_config under canarie-api

### PR DESCRIPTION
**Pull request by fmigneault**
## Changes

**Non-breaking changes**
- Add missing ``config/canarie-api/weaver_config.py`` entry to ``.gitignore`` of ``./components/weaver``
  that is generated from the corresponding template file.

**Breaking changes**
- n/a

## Related Issue / Discussion

- Resolves https://github.com/bird-house/birdhouse-deploy/pull/114#discussion_r741553883

### CODE REVIEW

**Review by matprov — APPROVED** [review]

**Review by tlvu — APPROVED** [review]

File: CHANGES.md [hunk 1]
```diff
@@ -14,7 +14,10 @@
 [Unreleased](https://github.com/bird-house/birdhouse-deploy/tree/master) (latest)
 ------------------------------------------------------------------------------------------------------------------
 
-[//]: # (list changes here, using '-' for each new entry, remove this when items are added)
+## Fixes
+
+- Add missing ``config/canarie-api/weaver_config.py`` entry to ``.gitignore`` of ``./components/weaver``
+  that is generated from the corresponding template file.
```
Comment (tlvu) [inline CHANGES.md:20]: @fmigneault add a note to this changelog that "Even with this autodeploy fix, autodeploy will not resume automatically.  A manual git pull will be required.  This fix is to fix subsequent autodeploy trigger only".

**Comment by fmigneault** [comment]
run tests

**Comment by fmigneault** [comment]
@matprov @tlvu 
FYI
All tests are passing, but following error is reported: 
http://daccs-jenkins.crim.ca:80/job/PAVICS-e2e-workflow-tests/job/master/656/

```
18:54:28  Archiving artifacts
18:54:30  ‘buildout/*.output.ipynb’ doesn’t match anything: ‘buildout’ exists but not ‘buildout/*.output.ipynb’
Error when executing always post condition:
hudson.AbortException: No artifacts found that match the file pattern "buildout/*.output.ipynb". Configuration error?
	at hudson.tasks.ArtifactArchiver.perform(ArtifactArchiver.java:280)
Caused: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:92)
	at org.jenkinsci.plugins.workflow.steps.CoreStep$Execution.run(CoreStep.java:70)
	at org.jenkinsci.plugins.workflow.steps.SynchronousNonBlockingStepExecution.lambda$start$0(SynchronousNonBlockingStepExecution.java:47)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by fmigneault** [comment]
@tlvu 
My bad. Forgot to add the PR description in the merge message again.

**Comment by tlvu** [comment]
> No artifacts found that match the file pattern "buildout/*.output.ipynb".

Yeah I noticed that recently too.  I am guessing due to PR https://github.com/Ouranosinc/PAVICS-e2e-workflow-tests/pull/91/files which puts each artifact into a different `archiveArtifacts` call.

Before all artifacts were in one single `archiveArtifacts` so as long as one is found no error is raised.

I am hoping there is an option to ignore missing artifact.  Else we just have to restore back to the one single `archiveArtifacts` call as before, which I agree is much less readable.

**Comment by fmigneault** [comment]
The is a `onlyIfSuccessful` parameter, but not much detail about it: https://www.jenkins.io/doc/pipeline/steps/core/

**Comment by fmigneault** [comment]
Nevermind: https://github.com/jenkinsci/jenkins/blob/master/core/src/main/java/hudson/tasks/ArtifactArchiver.java#L94-L97

**Comment by tlvu** [comment]
This is even  better https://github.com/jenkinsci/jenkins/blob/1d0e81acc820ea6531b33eac3a630dd10700e16a/core/src/main/java/hudson/tasks/ArtifactArchiver.java#L88-L92

**Pull request merged**